lib/livebook_web/plugs/configured_plug.ex

```diff
@@ -6,12 +6,13 @@ defmodule LivebookWeb.ConfiguredPlug do
   @behaviour Plug
 
   @impl true
-  def init(_opts) do
-    plugs = Application.get_env(:livebook, :plugs, [])
-    %{plugs: plugs}
-  end
+  def init(opts), do: opts
 
   @impl true
-  def call(conn, %{plugs: []}), do: conn
-  def call(conn, %{plugs: plugs}), do: Plug.run(conn, plugs)
+  def call(conn, _opts) do
+    case Application.get_env(:livebook, :plugs, []) do
```

Comment:
The default argument is unnecessary because we set it config/config.exs. However, when others use livebook as a dependency, none of our configurations are loaded, which means this is necessary.

There is actually a solution to this problem, which is to move our default values to:

    env: [plugs: []]

inside `def application`. Shall we start moving in this direction with this PR? Or should we open up a separate issue?

Reply:
I can open up a follow up PR :)

So we would move these values?

https://github.com/livebook-dev/livebook/blob/28b0ba9c4af086f055bcb8a21e429af10bec4f87/config/config.exs#L17-L34

The users will still need some of the configuration like `config :phoenix, :json_library, Jason` though.

Reply:
There is also an option to do [env: Application.get_all_env(:livebook)] and have all compile time config dumped there.

Reply:
Sounds good, any drawbacks?

Reply:
No, the more I think about it sounds like a good call. The only issue is that there is some default configuration that we set only on the config/runtime.exs... and I am not quite sure how to address it. For example, we load the filesystems only on the runtime, which means someone using Livebook as a dependency will get nothing.

One option is to move all of our config/runtime.exs to a public function so anyone can call it. I think @fhunleth has to manually figure out which variables are not set every time he starts it. :) Let's do this in a separate PR. For now I think it is enough to use `fetch_env!` here. We will discuss the rest separately.